src/poprox_recommender/pipeline.py

```diff
@@ -24,9 +26,10 @@ def __call__(self, inputs: dict[str, ArticleSet | InterestProfile]) -> ArticleSe
         # Avoid modifying the inputs
         state = deepcopy(inputs)
 
-        # Run each component in the order it was added
-        for component_spec in self.components:
-            state = self.run_component(component_spec, state)
+        with torch.inference_mode():
+            # Run each component in the order it was added
+            for component_spec in self.components:
+                state = self.run_component(component_spec, state)
```

Comment:
I don't know if this is the best place to put this, but we need to run the pipeline components in inference mode to avoid memory explosions.

Reply:
I see the issue this solves, but I also agree that this probably isn't the best place to put this. Ideally the pipeline abstraction has no knowledge of what frameworks the components are using, maybe the way to do this is to push it down into the components that use Torch?